sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Provider } from '../../providers/provider';
+import {
+  extractMutableReference,
+  extractStructTag,
+  getObjectReference,
+  isSharedObject,
+  isValidSuiAddress,
+  normalizeSuiObjectId,
+  SuiJsonValue,
+  SuiMoveNormalizedType,
+} from '../../types';
+import { bcs, CallArg, ObjectArg } from '../../types/sui-bcs';
+import { MoveCallTransaction } from './txn-data-serializer';
+
+const MOVE_CALL_SER_ERROR = 'Move call argument serialization error:';
+
+const isTypeFunc = (type: string) => (t: any) => typeof t === type;
+
+export class CallArgSerializer {
+  constructor(private provider: Provider) {}
+
+  async serializeMoveCallArguments(
+    txn: MoveCallTransaction
+  ): Promise<CallArg[]> {
+    const normalized = await this.provider.getNormalizedMoveFunction(
```

Comment:
This is a good thing to have built-in, but I think it'd also be ideal if we could at some point in the future auto-generate TS clients that could do this as well. It'd let us avoid an API call, and we could provide an even more ergonomic API for consumers.

Reply:
Agreed, this is on the roadmap